juniper/src/integrations/uuid.rs

```diff
@@ -1,28 +1,36 @@
 use uuid::Uuid;
 
+use parser::ParseError;
+use value::{ParseScalarValue, ScalarValue};
 use Value;
 
-graphql_scalar!(Uuid {
+graphql_scalar!(Uuid where Scalar = <S> {
```

Comment:
I really don't like this public API. Is it possible to default to `<S>` when there is not a `where Scaler =` in the macro?

Reply:
It already defaults to `<__S>` but in my opinion (at least in this case) it is more readable if there is no type parameter coming from somewhere.

In cases where someone want's to define a new scalar type based on an existing one, the `where Scalar = …` clause is not necessary, because the `from_str` function could be written in terms of the existing type (for example `<String as ParseScalarValue<_>>::from_str(value)`).

For cases where a custom scalar type is used the scalar type needs to be specified explicitly.

Reply:
> Is it possible to default to <S> when there is not a where Scaler = in the macro?

In theory this should be possible, but then there would be a generic parameters coming from somewhere. In my opinion that's noting we should do .
Currently the default is set to `DefaultScalarValue`. All other variants need some syntax to specify them. If someone has a better idea than `where Scalar = …`, that should be quite easy to change.)